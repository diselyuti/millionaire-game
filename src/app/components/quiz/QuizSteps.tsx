import React from 'react';
import QuizStepInactive from '@/app/components/quiz/QuizStepInactive.tsx';
import QuizStepFinished from '@/app/components/quiz/QuizStepFinished.tsx';
import QuizStepSelected from '@/app/components/quiz/QuizStepSelected.tsx';
import styles from '@/app/styles/components/quiz/quiz-step.module.scss';
import { IStep } from '@/types/IStep.ts';

function QuizSteps({ steps }: { steps: IStep[] }) {
  return (
    <div className={styles.steps}>
      {steps.map((step) => {
        if (step.finished) {
          return (
            <QuizStepFinished
              key={step.id}
              text={`$${step.prize.toLocaleString()}`}
            />
          );
        }
        if (step.selected) {
          return (
            <QuizStepSelected
              key={step.id}
              text={`$${step.prize.toLocaleString()}`}
            />
          );
        }
        return (
          <QuizStepInactive
            key={step.id}
            text={`$${step.prize.toLocaleString()}`}
          />
        );
      })}
    </div>
  );
}

export default QuizSteps;
