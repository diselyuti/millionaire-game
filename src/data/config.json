{
  "currentQuestion": 0,
  "steps": [
    {
      "id": 1,
      "prize": 100,
      "selected": false,
      "finished": false
    },
    {
      "id": 2,
      "prize": 200,
      "selected": false,
      "finished": false
    },
    {
      "id": 3,
      "prize": 300,
      "selected": false,
      "finished": false
    },
    {
      "id": 4,
      "prize": 500,
      "selected": false,
      "finished": false
    },
    {
      "id": 5,
      "prize": 1000,
      "selected": false,
      "finished": false
    },
    {
      "id": 6,
      "prize": 2000,
      "selected": false,
      "finished": false
    },
    {
      "id": 7,
      "prize": 4000,
      "selected": false,
      "finished": false
    },
    {
      "id": 8,
      "prize": 8000,
      "selected": false,
      "finished": false
    }
  ],
  "isGameOver": false,
  "questions": [
    {
      "question": "Якими кольорами складається прапор України?",
      "correctAnswerId": 1,
      "prizeId": 1,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Жовтий та синій"
        },
        {
          "id": 2,
          "answer": "Червоний та чорний"
        },
        {
          "id": 3,
          "answer": "Зелений та блакитний"
        },
        {
          "id": 4,
          "answer": "Червоний та білий"
        }
      ]
    },
    {
      "question": "Який океан є найбільшим за площею?",
      "correctAnswerId": 2,
      "prizeId": 2,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Атлантичний океан"
        },
        {
          "id": 2,
          "answer": "Тихий океан"
        },
        {
          "id": 3,
          "answer": "Індійський океан"
        },
        {
          "id": 4,
          "answer": "Північний Льодовитий океан"
        }
      ]
    },
    {
      "question": "Хто написав роман \"Майстер і Маргарита\"?",
      "correctAnswerId": 3,
      "prizeId": 3,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Лев Толстой"
        },
        {
          "id": 2,
          "answer": "Федор Достоєвський"
        },
        {
          "id": 3,
          "answer": "Михайло Булгаков"
        },
        {
          "id": 4,
          "answer": "Антон Чехов"
        }
      ]
    },
    {
      "question": "Який материк має найбільшу площу?",
      "correctAnswerId": 3,
      "prizeId": 4,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Африка"
        },
        {
          "id": 2,
          "answer": "Австралія"
        },
        {
          "id": 3,
          "answer": "Євразія"
        },
        {
          "id": 4,
          "answer": "Північна Америка"
        }
      ]
    },
    {
      "question": "Як називається столиця Японії?",
      "correctAnswerId": 3,
      "prizeId": 5,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Киото"
        },
        {
          "id": 2,
          "answer": "Осака"
        },
        {
          "id": 3,
          "answer": "Токіо"
        },
        {
          "id": 4,
          "answer": "Хоккайдо"
        }
      ]
    },
    {
      "question": "Хто зображений на найпоширеніших банкнотах долара США?",
      "correctAnswerId": 2,
      "prizeId": 6,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Джордж Вашингтон"
        },
        {
          "id": 2,
          "answer": "Авраам Лінкольн"
        },
        {
          "id": 3,
          "answer": "Бенджамін Франклін"
        },
        {
          "id": 4,
          "answer": "Томас Джефферсон"
        }
      ]
    },
    {
      "question": "Як називається найвища гора світу?",
      "correctAnswerId": 2,
      "prizeId": 7,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Монблан"
        },
        {
          "id": 2,
          "answer": "Еверест"
        },
        {
          "id": 3,
          "answer": "Кіліманджаро"
        },
        {
          "id": 4,
          "answer": "Аконкагуа"
        }
      ]
    },
    {
      "question": "Як називається фігура з чотирма однаковими сторонами і чотирма прямими кутами?",
      "correctAnswerId": 4,
      "prizeId": 8,
      "isAnswered": false,
      "answers": [
        {
          "id": 1,
          "answer": "Круг"
        },
        {
          "id": 2,
          "answer": "Трикутник"
        },
        {
          "id": 3,
          "answer": "Прямокутник"
        },
        {
          "id": 4,
          "answer": "Квадрат"
        }
      ]
    }
  ]
}
