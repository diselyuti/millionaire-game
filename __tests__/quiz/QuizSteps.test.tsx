import React from 'react';
import '@testing-library/jest-dom';
import {render} from "@testing-library/react";
import QuizSteps from "@/app/components/quiz/QuizSteps.tsx";
import config from "@/data/config.json";
describe('QuizSteps', () => {
  test('renders component correctly', () => {
    const { getByText } = render(<QuizSteps steps={config.steps}/>);
    expect(getByText('$100')).toBeInTheDocument();
    expect(getByText('$200')).toBeInTheDocument();
    expect(getByText('$300')).toBeInTheDocument();
    expect(getByText('$500')).toBeInTheDocument();
    expect(getByText('$1,000')).toBeInTheDocument();
    expect(getByText('$2,000')).toBeInTheDocument();
    expect(getByText('$4,000')).toBeInTheDocument();
    expect(getByText('$8,000')).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { container } = render(<QuizSteps steps={config.steps}/>);
    expect(container).toMatchSnapshot();
  });
});
