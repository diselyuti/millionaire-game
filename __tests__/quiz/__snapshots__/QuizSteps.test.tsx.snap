// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizSteps should match snapshot 1`] = `
<div>
  <div
    class="steps"
  >
    <article
      class="step"
    >
      <p>
        $100
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $200
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $300
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $500
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $1,000
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $2,000
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $4,000
      </p>
    </article>
    <article
      class="step"
    >
      <p>
        $8,000
      </p>
    </article>
  </div>
</div>
`;
